lib/task-service.js

```diff
@@ -4,21 +4,54 @@
 
 var di = require('di');
 
-module.exports = factory;
-di.annotate(factory, new di.Provide('Services.Task'));
-di.annotate(factory, new di.Inject(
+module.exports = taskServiceFactory;
+di.annotate(taskServiceFactory, new di.Provide('Services.Task'));
+di.annotate(taskServiceFactory, new di.Inject(
+    'Logger',
+    'Promise',
+    'ChildProcess',
     'TaskOption.Validator'
 ));
 
-function factory(validator) {
+function taskServiceFactory(
+    Logger,
+    Promise,
+    ChildProcess,
+    validator
+) {
+    var logger = Logger.initialize(taskServiceFactory);
+    var requiredCmdTools = ['ipmitool'];
```

Comment:
@yyscamper I think this check is logically misplaced in `taskService.start`.  Tasks and workflows are dependent on tools, not the service itself.  Rather than preventing the service from starting if tools are missing, I think it would be better to check for existence of all tools needed to run a graph prior to starting it.